apps/app_utf8.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <openssl/buffer.h>
+#include <openssl/asn1.h>        /* For UTF8_putc */
+#include "apps.h"
+
+char *to_utf8(const char *str)
+{
+    char *str_utf8 = NULL;
+
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199409L \
```

Comment:
While this is formally dandy, there is a bunch of compilers that are somewhat lax about this. In sense that there is mbstowcs, yet __STDC_VERSION__ is not defined, at least not by default. Simplest example is gcc 4.8, which still is RedHat's choice for their Enterprise distros. I mean if you just run ./config on RedHat 7.5, the latest, you'll miss this code path. In other words suggestion is to complement condition with at least || defined(__linux). And even though operator precedence is && over ||, I'd like to suggest to enclose the above condition into (). One can pose same question about other platforms...